store/postgres_graphql.go

```diff
@@ -116,20 +116,26 @@ func psqlResolveRootQuery(pool *pgxpool.Pool, tenant string, graph *SchemaGraph,
 	if err != nil {
 		return nil, fmt.Errorf("failed to execute SQL query: %s: %w", sqlStr, err)
 	}
-
 	defer rows.Close()
 
 	// Iterate through the result set and append each row of results to the
-	// result value we are returning
+	// result value we are returning. We should check if there are no rows
+	// in which case we want to return at least an empty slice
+	var hasRows bool
 	for rows.Next() {
+		hasRows = true
 		if err := psqlScanRowColumns(rows, result, rootColumns); err != nil {
 			return nil, fmt.Errorf("failed scanning row values: %w", err)
 		}
 	}
+	if !hasRows {
+		// Initialize with an empty slice to avoid returning just null
+		result[rootTable] = make([]interface{}, 0)
```

Comment:
Can this not be done _before_ entering the loop? I presume if the code inside the loop runs, the `result[rootTable]` would be re-initialised or replaced anyway?

Reply:
Nice suggestion! Fixed